beacon-chain/forkchoice/protoarray/proposer_boost.go

```diff
@@ -0,0 +1,71 @@
+package protoarray
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/config/params"
+	"github.com/prysmaticlabs/prysm/time/slots"
+)
+
+// BoostProposerRoot sets the block root which should be boosted during
+// the LMD fork choice algorithm calculations. This is meant to reward timely,
+// proposed blocks which occur before a cutoff interval set to
+// SECONDS_PER_SLOT // INTERVALS_PER_SLOT.
+//
+//  time_into_slot = (store.time - store.genesis_time) % SECONDS_PER_SLOT
+//  is_before_attesting_interval = time_into_slot < SECONDS_PER_SLOT // INTERVALS_PER_SLOT
+//  if get_current_slot(store) == block.slot and is_before_attesting_interval:
+//      store.proposer_boost_root = hash_tree_root(block)
+func (f *ForkChoice) BoostProposerRoot(_ context.Context, blockSlot types.Slot, blockRoot [32]byte, genesisTime time.Time) error {
```

Comment:
```suggestion
func (f *ForkChoice) BoostProposerRoot(_ context.Context, blockSlot types.Slot, blockRoot [32]byte) error {
```

Nit. Forkchoice store should just track genesis time. We don't have to fix this in this PR. It can be an excellent first issue

Reply:
This is a harder issue to solve as it would require more refactoring than I thought, so yes let's have it as a good first issue

Reply:
Added https://github.com/prysmaticlabs/prysm/issues/10099